src/App.tsx

```diff
@@ -1,24 +1,214 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { Filter, MakeTodosCompleted, Todo } from './types/Todo';
+import * as todosService from './api/todos';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
+import { TempTodo } from './components/TempTodo';
 
-const USER_ID = 0;
+const USER_ID = 11641;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [todoTitle, setTodoTitle] = useState('');
+  const [filterBy, setFilterBy] = useState<Filter>(Filter.all);
+  const [hasCompleted, setHasCompleted] = useState(false);
+  const [quantity, setQuantity] = useState(0);
+  const [hideNotification, setHideNotification] = useState(true);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isFocused, setIsFocused] = useState(true);
+  const [allTodosCompleted, setAllTodosCompleted] = useState(false);
+  const [makeTodosComplete, setMakeTodosComplete]
+  = useState(MakeTodosCompleted.begin);
+  const [clearCompleted, setClearCompleted] = useState(false);
+
+  useEffect(() => {
+    const filteredTodos = todos.filter((todo) => todo.completed);
+
+    setQuantity(todos.length - filteredTodos.length);
+  }, [todos]);
+
+  useEffect(() => {
+    setErrorMessage('');
+
+    todosService
+      .getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+      });
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    const filteredByTodos = todos.length
+      ? todos.filter((todo) => {
+        switch (filterBy) {
+          case Filter.active:
+            return !todo.completed;
+          case Filter.completed:
+            return todo.completed;
+          default:
+            return true;
+        }
+      })
+      : [];
+
+    return filteredByTodos;
+  }, [filterBy, todos]);
+
+  const deleteTodo = (todoId: number) => {
+    setErrorMessage('');
+    setIsFocused(false);
+
+    return todosService
+      .deleteTodo(todoId)
+      .then(() => {
+        setTodos((currentTodos) => currentTodos
+          .filter((todo) => todo.id !== todoId));
+      })
+      .catch((error) => {
+        setErrorMessage('Unable to delete a todo');
+        throw error;
+      })
+      .finally(() => {
+        setIsFocused(true);
+      });
+  };
```

Comment:
consider to use async/await instead of then/catch. it is more readable imho
```suggestion
  const deleteTodo = async (todoId: number) => {
  setErrorMessage('');
  setIsFocused(false);

  try {
    await todosService.deleteTodo(todoId);
    const updatedTodos = currentTodos => currentTodos.filter(todo => todo.id !== todoId);
    setTodos(updatedTodos);
  } catch (error) {
    setErrorMessage('Unable to delete a todo');
  } finally {
    setIsFocused(true);
  }
};
```